### DISCUSSION TITLE
import libraries: is it cheating ?

### DISCUSSION THREAD

**Discussion by AnnaMilena** [codewars/codewars.com#3175] [General]
I see some kata for arrays , regular expressions , is it cheating, if I import libraries in python to solve these kata?

**Comment by hobovsky** [comment]
No, using available libraries usually is not cheating. Exception are kata where description says something like _"`numpy` is blocked and you cannot use it"_, and tests block the use of the module - in such case, working around the restriction can be considered cheating. But if descritpion does not disallow any libraries and tests do not block them, then users are allowed to use available libraires.

**Comment by bestlim** [comment]
nah
